hack/update-codegen.sh

```diff
@@ -74,36 +80,52 @@ function generate_group() {
     --internal-clientset-package ${CLIENT_PKG}/clientset_generated/internalclientset \
     --listers-package ${LISTERS_PKG} \
     --output-package ${INFORMERS_PKG} \
-    --output-base ${GEN_TMPDIR}
+    --output-base "${OUTPUT_BASE}" \
+    "$4"
 
   echo "generating deep copies"
   bazel run //vendor/k8s.io/code-generator/cmd/deepcopy-gen -- \
     --go-header-file "${SCRIPT_ROOT}/boilerplate/boilerplate.go.txt" \
     --input-dirs ${APIS_PKG}/${GROUP_NAME},${APIS_PKG}/${GROUP_NAME}/${VERSION} \
-    --output-base ${GEN_TMPDIR} \
-    --output-file-base zz_generated.deepcopy
+    --output-base "${OUTPUT_BASE}" \
+    --output-file-base zz_generated.deepcopy \
+    "$4"
 
   echo "generating defaults"
   bazel run //vendor/k8s.io/code-generator/cmd/defaulter-gen -- \
     --go-header-file "${SCRIPT_ROOT}/boilerplate/boilerplate.go.txt" \
     --input-dirs ${APIS_PKG}/${GROUP_NAME},${APIS_PKG}/${GROUP_NAME}/${VERSION} \
-    --output-base ${GEN_TMPDIR} \
-    --output-file-base zz_generated.defaults
+    --output-base "${OUTPUT_BASE}" \
+    --output-file-base zz_generated.defaults \
+    "$4"
 
   echo "generating conversions"
   bazel run //vendor/k8s.io/code-generator/cmd/conversion-gen -- \
     --go-header-file "${SCRIPT_ROOT}/boilerplate/boilerplate.go.txt" \
     --input-dirs ${APIS_PKG}/${GROUP_NAME},${APIS_PKG}/${GROUP_NAME}/${VERSION} \
     --extra-peer-dirs "k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/conversion,k8s.io/apimachinery/pkg/runtime" \
-    --output-base ${GEN_TMPDIR} \
-    --output-file-base zz_generated.conversion
-
-  cp -r "${GEN_TMPDIR}/k8s.io/cluster-registry/"* "${SCRIPT_BASE}/${REPO_DIRNAME}"
+    --output-base "${OUTPUT_BASE}" \
+    --output-file-base zz_generated.conversion \
+    "$4"
 }
 
+# Set up the temporary GOPATH with necessary dependencies.
 mkdir -p "${TMP_GOPATH}/src/k8s.io/cluster-registry"
 mkdir -p "${TMP_GOPATH}/src/k8s.io/apimachinery"
 cp -r "${SCRIPT_ROOT}/../"* "${TMP_GOPATH}/src/k8s.io/cluster-registry"
 cp -r "${SCRIPT_ROOT}/../vendor/k8s.io/apimachinery/"* "${TMP_GOPATH}/src/k8s.io/apimachinery"
```

Comment:
I know this is from a past commit, but what's the reason for copying over apimachinery into the temporary GOPATH?

Reply:
`client-gen` requires that the apimachinery repo be in the GOPATH. I'm not sure exactly why that dependency exists, but client generation fails without some packages in `apimachinery`.

Reply:
Ah ok, thanks for the explanation. Maybe just add a comment about that next time you're in this area.